04_predict_controls/nbconverted/plot_msi_results.py

```diff
@@ -0,0 +1,179 @@
+axarr[1].set_title('MSI prediction performance, AUROC')
+axarr[1].set_xlabel('Cancer type')
+axarr[1].set_ylabel('AUROC')
+axarr[1].set_ylim(0.35, 1.05)
+axarr[1].legend(title='Data type')
+
+
+# ### Plot results per cancer type, corrected for permuted labels baseline
+
+# In[5]:
+
+
+compare_df = au.compare_all_data_types(results_df,
+                                       sig_alpha=0.05,
+                                       filter_genes=False,
+                                       compare_ind=True,
+                                       metric='auroc')
+compare_df.rename(columns={'gene': 'cancer_type'}, inplace=True)
+
+compare_aupr_df = au.compare_all_data_types(results_df,
+                                            sig_alpha=0.05,
+                                            filter_genes=False,
+                                            compare_ind=True,
+                                            metric='aupr')
+compare_aupr_df.rename(columns={'gene': 'cancer_type'}, inplace=True)
+
+print(compare_df.shape)
+print(compare_df.training_data.unique())
+compare_df.head()
+
+
+# In[6]:
+
+
+sns.set({'figure.figsize': (14, 12)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(2, 1)
+
+sns.boxplot(data=compare_aupr_df, x='cancer_type', y='delta_aupr',
+            hue='training_data',
+            hue_order=sorted(compare_aupr_df.training_data.unique()),
+            ax=axarr[0])
+axarr[0].set_title('MSI prediction performance, AUPR vs. permuted baseline')
+axarr[0].set_xlabel('Cancer type')
+axarr[0].set_ylabel('AUPR(signal) - AUPR(shuffled)')
+axarr[0].set_ylim(0.0, 1.0)
+axarr[0].legend(title='Data type')
+
+sns.boxplot(data=compare_df, x='cancer_type', y='delta_auroc',
+            hue='training_data',
+            hue_order=sorted(compare_df.training_data.unique()),
+            ax=axarr[1])
+axarr[1].set_title('MSI prediction performance, AUROC vs. permuted baseline')
+axarr[1].set_xlabel('Cancer type')
+axarr[1].set_ylabel('AUROC(signal) - AUROC(shuffled)')
+axarr[1].set_ylim(0.0, 1.0)
+axarr[1].legend(title='Data type')
+
+
+# ### Look at features selected from mutational signatures data
+# 
+# There are a few mutational signatures that have been directly associated with MSI. Let's see if those are selected frequently in our models, and if they tend to have positive coefficients (i.e. higher prevalence of signature -> higher probability of MSI-H status).
+
+# In[7]:
+
+
+nz_coefs = []
+for identifier, seed, coefs in au.generate_nz_coefs_msi(results_dir):
+    if 'mut_sigs' not in identifier: continue
+    for fold_no, fold_coefs in enumerate(coefs):
+        for feat_name, feat_coef in fold_coefs:
+            nz_coefs.append([identifier, seed, fold_no, feat_name, feat_coef])
+            
+nz_coefs_df = pd.DataFrame(nz_coefs, columns=['identifier', 'seed', 'fold', 'name', 'coef'])
+nz_coefs_df.head()
+
+
+# In[8]:
+
+
+# just average each coefficient over seeds/folds for now
```

Comment:
Can you compare the coeff/weight from your regression model across your different cancer models? I guess you can at least say if they are in the upper quantile of the distribution of weights for your model..

Reply:
Yeah, I think this would be interesting to dive deeper into. I think for this paper we're not interested in going too far into how the feature selection works (other than the basic sanity checks in this NB), but after this paper is submitted I'm tentatively planning to look further into some of these things as a start to my next project. Stay tuned...